__tests__/distributors/base-installer.test.ts

```diff
@@ -0,0 +1,210 @@
+    [{ version: '11', arch: 'x86', packageType: 'jre' }, null]
+  ])(`should find java for path %s -> %s`, (input, expected) => {
+    mockJavaBase = new EmptyJavaBase(input);
+    expect(mockJavaBase['findInToolcache']()).toEqual(expected);
+  });
+});
+
+describe('setupJava', () => {
+  const actualJavaVersion = '11.1.10';
+  const javaPath = path.join('Java_Empty_jdk', actualJavaVersion, 'x86');
+
+  let mockJavaBase: EmptyJavaBase;
+
+  let spyTcFind: jest.SpyInstance;
+  let spyCoreDebug: jest.SpyInstance;
+  let spyCoreInfo: jest.SpyInstance;
+  let spyCoreExportVariable: jest.SpyInstance;
+  let spyCoreAddPath: jest.SpyInstance;
+  let spyCoreSetOutput: jest.SpyInstance;
+
+  beforeEach(() => {
+    spyTcFind = jest.spyOn(tc, 'find');
+    spyTcFind.mockImplementation((toolname: string, javaVersion: string, architecture: string) => {
+      const semverVersion = new semver.Range(javaVersion);
+
+      if (path.basename(javaPath) !== architecture || !javaPath.includes(toolname)) {
+        return '';
+      }
+
+      return semver.satisfies(actualJavaVersion, semverVersion) ? javaPath : '';
+    });
+
+    // Spy on core methods
+
+    spyCoreDebug = jest.spyOn(core, 'debug');
+    spyCoreDebug.mockImplementation(() => undefined);
+
+    spyCoreInfo = jest.spyOn(core, 'info');
+    spyCoreInfo.mockImplementation(() => undefined);
+
+    spyCoreAddPath = jest.spyOn(core, 'addPath');
+    spyCoreAddPath.mockImplementation(() => undefined);
+
+    spyCoreExportVariable = jest.spyOn(core, 'exportVariable');
+    spyCoreExportVariable.mockImplementation(() => undefined);
+
+    spyCoreSetOutput = jest.spyOn(core, 'setOutput');
+    spyCoreSetOutput.mockImplementation(() => undefined);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '11', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ],
+    [
+      { version: '11.1', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ],
+    [
+      { version: '11.1.10', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ]
+  ])('should find java for path %s -> %s', (input, expected) => {
+    mockJavaBase = new EmptyJavaBase(input);
+    expect(mockJavaBase.setupJava()).resolves.toEqual(expected);
+    expect(spyTcFind).toHaveBeenCalled();
+  });
+
+  it.each([
+    [
+      { version: '11', arch: 'x86', packageType: 'jre' },
+      { javaPath: `/toolcache/Java_Empty_jre/11.0.8`, javaVersion: '11.0.8' }
+    ],
+    [
+      { version: '11', arch: 'x64', packageType: 'jdk' },
+      { javaPath: `/toolcache/Java_Empty_jdk/11.0.8`, javaVersion: '11.0.8' }
+    ],
+    [
+      { version: '11', arch: 'x64', packageType: 'jre' },
+      { javaPath: `/toolcache/Java_Empty_jre/11.0.8`, javaVersion: '11.0.8' }
+    ]
+  ])('download java', async (input, expected) => {
```

Comment:
please, add parameters to test description, otherwise it is not obvious what exact configuration failed when all 3 tests are called "download java"

Reply:
Done. Thank you.